symforce/slam/imu_preintegration/imu_preintegrator.h

```diff
@@ -0,0 +1,78 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <Eigen/Core>
+
+#include <sym/util/epsilon.h>
+
+#include "./preintegrated_imu_measurements.h"
+
+namespace sym {
+
+/**
+ * Class to on-manifold preintegrate IMU measurements for usage in a SymForce optimization
+ * problem.
+ *
+ * For usage, see the doc-string of ImuFactor in imu_factor.h
+ */
+template <typename Scalar>
+class ImuPreintegrator {
+ public:
+  using Vector3 = typename PreintegratedImuMeasurements<Scalar>::Vector3;
+  using Matrix33 = typename PreintegratedImuMeasurements<Scalar>::Matrix33;
+  using Matrix99 = Eigen::Matrix<Scalar, 9, 9>;
+
+ private:
+  PreintegratedImuMeasurements<Scalar> preintegrated_measurments_;
+  Matrix99 covariance_;  // covariance of [DR.ToTangent(), Dv, Dp]
```

Comment:
Yeah, you're right. When I first wrote it I wanted to convey the order of the arguments, but didn't have an immediate idea on how to succinctly describe what about the rotation we were taking the covariance of, so I used this as an approximation.

Still, I don't think it's good to let it stand as I can imagine it confusing someone and wasting their time as they try to check the math.

One idea is to say
`covariance of [DR, Dv, Dp] in local coordinates of mean`
(kind of awkward)

I also noticed that elsewhere I wrote `covariance of [DR's tangent space, Dv, Dp]`, but that's not very precise. I think I'll go with the former.

I've made the change.

Reply:
SGTM, I'm also good with `covariance of [DR's tangent space, Dv, Dp]` as it's much more concise and generally understood to mean the more precise version.  It would also be fine to just say `covariance of [DR, Dv, Dp]`, since the covariance of a lie group element is understood to be on the tangent space at the mean